builtin/bisect--helper.c

```diff
@@ -389,6 +394,219 @@ static int bisect_terms(struct bisect_terms *terms, const char *option)
 	return 0;
 }
 
+static int bisect_append_log_quoted(const char **argv)
+{
+	int res = 0;
+	FILE *fp = fopen(git_path_bisect_log(), "a");
+	struct strbuf orig_args = STRBUF_INIT;
+
+	if (!fp)
+		return -1;
+
+	if (fprintf(fp, "git bisect start") < 1) {
+		res = -1;
+		goto finish;
+	}
+
+	sq_quote_argv(&orig_args, argv);
+	if (fprintf(fp, "%s\n", orig_args.buf) < 1)
+		res = -1;
+
```

Comment:
When there is only one `goto` it is often possible and worth it to avoid the goto.

For example:

```
	if (fprintf(fp, "git bisect start") < 1) {
		res = -1;
	} else {
		sq_quote_argv(&orig_args, argv);
		if (fprintf(fp, "%s\n", orig_args.buf) < 1)
			res = -1;
	}
```

Reply:
@chriscool Why don't you append a `c` to the <pre>```</pre> line, to add syntax coloring?

Reply:
Done.

Reply:
this is from commit https://github.com/gitgitgadget/git/commit/06f5608c14e64e6 which is already in `tt/bisect-in-c`, is this change worth having its own commit?

Reply:
Feel free to skip this change if you want. It's not a big deal.